src/server/pps/server/api_server.go

```diff
@@ -4050,3 +4052,39 @@ func (a *apiServer) SetProjectDefaults(ctx context.Context, req *pps.SetProjectD
 	}
 	return &resp, nil
 }
+
+// GetLogsV2 request is handled by apiServer.logsReqHandler that returns a channel that receives log lines that meet request parameters
+func (a *apiServer) GetLogsV2(request *pps.GetLogsV2Request, apiGetLogsV2Server pps.API_GetLogsV2Server) (retErr error) {
+
+	ctx := apiGetLogsV2Server.Context()
+
+	getLogsV2Handler := &getLogsV2Handler{
+		apiServer: a,
+	}
+	logsChannel, err := getLogsV2Handler.handleRequest(ctx, request)
+	if logsChannel == nil {
+		return errors.EnsureStack(err)
+	}
+	jp := &protojson.UnmarshalOptions{
```

Comment:
This reminds me that for V1 loki logs, we have to "guess" loki's underlying log format because people configure that wrong (loki needs to know whether the user is using docker, cri-o, or containerd as the cluster's container engine... if you guess wrong, our log lines are not JSON, but rather some other format; see the code for details).  I think that we probably still need that code, as we didn't change anyone's loki configuration.